xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-test/xwiki-platform-notifications-test-pageobjects/src/main/java/org/xwiki/platform/notifications/test/po/GroupedNotificationElementPage.java

```diff
@@ -0,0 +1,113 @@
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.platform.notifications.test.po;
+
+import java.util.List;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.xwiki.test.ui.XWikiWebDriver;
+import org.xwiki.test.ui.po.BaseElement;
+
+import static org.openqa.selenium.By.cssSelector;
+
+/**
+ * Represent generic operations on the notification tray groups.
+ *
+ * @version $Id$
+ * @since 12.8RC1
+ */
+public class GroupedNotificationElementPage extends BaseElement
+{
+    protected static final String NOTIFICATION_EVENT_SELECTOR = "div.notification-event";
+
+    protected static final String TEXT_SELECTOR = "tr td.description";
+
+    private static final String GROUP_SELECTOR =
+        "button.toggle-notification-event-details";
+
+    private static final String EMITTER_SELECTOR =
+        "tr td span.notification-event-user";
+
+    private final WebElement rootElement;
+
+    /**
+     * Default constructor.
+     * @param rootElement The root element of the notification blocks.
+     */
+    public GroupedNotificationElementPage(WebElement rootElement)
+    {
+
+        this.rootElement = rootElement;
+    }
+
+    /**
+     * Opens a notification group.
+     * @param notificationNumber The notification index.
+     */
+    public void openGroup(int notificationNumber)
+    {
+        WebElement notifications =
+            findNotifications().get(notificationNumber);
+        WebElement group = notifications.findElement(cssSelector(GROUP_SELECTOR));
+        group.click();
+    }
+
+    /**
+     * Finds the text of a notification.
+     * @param notificationNumber The notification group number.
+     * @param groupIndex The index of the notification in the group.
+     * @return The mention text.
+     */
+    public String getText(int notificationNumber, int groupIndex)
+    {
+        WebElement notifications = findNotifications().get(notificationNumber);
+        WebElement element = notifications.findElements(cssSelector(TEXT_SELECTOR)).get(groupIndex);
+        return element.getText();
+    }
+
+    /**
+     *
+     * @param notificationNumber The notification group number.
+     * @param groupIndex The index of the notification in the group.
+     * @return The mention emitter.
+     */
+    public String getEmitter(int notificationNumber, int groupIndex)
+    {
+        WebElement notifications = findNotifications().get(notificationNumber);
+        WebElement emitter = notifications.findElements(cssSelector(EMITTER_SELECTOR)).get(groupIndex);
+        return emitter.findElement(By.tagName("a")).getText();
+    }
+
+    /**
+     *
+     * @param driver XWIki web driver.
+     * @return The root element of the notification tray.
+     */
+    public static WebElement getRootElementNotificationTray(XWikiWebDriver driver)
```

Comment:
I wouldn't have done it like this: we already have a PO for the Notification Tray so it should be its responsibility to create a GroupedNotificationElementPage with the right parent IMO.